### ISSUE TITLE
Smart colon height adjustment OT feature

### ISSUE THREAD

**Issue by chrissimpkins** [googlefonts/googlesans#63]
Request from @davelab6 / internal design team (contact ryankiley@)

> SF has some 'liga' code which uses GSUB classes to only apply between {0-9}{0-9}:{0-9}{0-9} (excuse the pseudo regex ;) and maybe it is worthwhile to do something similar here to automatically raise the colon to look centered for clock times.
>
>Possibly this should also go in the case feature for {A-Z}:{\ } etc

Support is currently available with `colon.time` design via SS02.  This request is to broaden the feature support.

**Comment by chrissimpkins** [comment]
Related https://github.com/razmadzekoko/google-sans/issues/2 where @razmadzekoko and @gorjious are discussing colon position + case feature support.

**Comment by chrissimpkins** [comment]
@Colophon-Foundry Edd, re: the request from a Google team here, what are your thoughts about support in `calt` and `case`?

**Comment by chrissimpkins** [comment]
Example from Akaki

<img width="893" alt="68747470733a2f2f33773070386270756b33743375783775653134736432726c7a622d7770656e67696e652e6e6574646e612d73736c2e636f6d2f77702d636f6e74656e742f75706c6f6164732f323031352f31302f766572746963616c6c792d63656e74657265642d636f6c6f6e2e706e67" src="https://user-images.githubusercontent.com/4249591/86262171-8ab32300-bb8d-11ea-9bc7-ce9d58c7cbbd.png">

**Comment by razmadzekoko** [comment]
We have similar case in terms of `hyphen`
there is used case feature, the glyph name is `hyphen.cap`
<img width="739" alt="Screen Shot 2020-07-01 at 19 30 18" src="https://user-images.githubusercontent.com/18236719/86262818-6d07ad00-bbd1-11ea-9040-bc6cdfe9fa7d.png">
So the similar way should be used for colon, I believe. 
I mean: `colon` and `colon.cap`

**Comment by chrissimpkins** [comment]
Edd, Akaki, Gor and I discussed this in a meeting today.  We decided that we will implement the adjusted cap height centered colon position between numerals in a `calt` feature.  

We will discuss the default colon position adjacent to upper case alphabetics with Jonathan Lee and Damien Correll.  This is an issue that impacts other scripts (e.g., the all caps forms in Georgian).  We reviewed the issue very briefly in SF during the meeting and it appears that the colon is set centered on x-height with English phrases that use colon near upper case.  More review required to determine approach for e.g., spreadsheet column/row indicators like `A:2`.

**Comment by Colophon-Foundry** [comment]
Commit  687baf3 in the [Smart Colon branch](https://github.com/googlefonts/googlesans/tree/smart-colon) adds support for this issue.

The code below details the substitutions.

```
sub [zero one two three four five six seven eight nine] colon' [zero one two three four five six seven eight nine] by colon.time;
sub [zero.cap one.cap two.cap three.cap four.cap five.cap six.cap seven.cap eight.cap nine.cap zero.alt.cap one.alt.cap six.alt.cap seven.alt.cap nine.alt.cap] colon' [zero.cap one.cap two.cap three.cap four.cap five.cap six.cap seven.cap eight.cap nine.cap zero.alt.cap one.alt.cap six.alt.cap seven.alt.cap nine.alt.cap] by colon.time;
sub [zero.tf one.tf two.tf three.tf four.tf five.tf six.tf seven.tf eight.tf nine.tf] colon' [zero.tf one.tf two.tf three.tf four.tf five.tf six.tf seven.tf eight.tf nine.tf] by colon.time;
sub [zero.ss03 one.ss03 two.ss03 three.ss03 four.ss03 five.ss03 six.ss03 seven.ss03 eight.ss03 nine.ss03] colon' [zero.ss03 one.ss03 two.ss03 three.ss03 four.ss03 five.ss03 six.ss03 seven.ss03 eight.ss03 nine.ss03] by colon.time;
```

This will cover the following use cases:
number : number
number.cap : number.cap
number.alternate.cap : number.alternate.cap
number.tf : number:tf
number.ss03 : number.ss03 **note: only in Uprights**

**Comment by Colophon-Foundry** [comment]
Forgot to mention. The above code has been added into the `calt` feature

**Comment by chrissimpkins** [comment]
Thanks Edd!

**Comment by chrissimpkins** [comment]
Changes added in #72. The colon positioning adjustments will be supported in a `calt` feature and released as part of the v3.001 work
